test/RSKAddressValidator.test.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  TestRSKAddressValidatorInstance
+} from '../types/truffle-contracts'
+import { toChecksumAddress } from 'ethereumjs-util'
+
+const TestRSKAddressValidator = artifacts.require('TestRSKAddressValidator')
+
+contract('RSKAddressValidator', function (accounts) {
+  let addressValidator: TestRSKAddressValidatorInstance
+  it('should return true on check with data signed with zero', async function () {
+    const messageHash = '0xf7cf90057f86838e5efd677f4741003ab90910e4e2736ff4d7999519d162d1ed'
+
+    const v = 27
+    const r = '3fba4230c7b08aa69affd0b7425b8666eb5e7d36f9ef84efeee5ba612c8df0bc'
+    const s = '6b212d83adb8364e5519a051894f4a513b7eb68d087d710bf21e70775d31a97b'
+
+    const signature = `0x${r}${s}${v.toString(16)}`
+
+    addressValidator = await TestRSKAddressValidator.new()
+    const addr = await addressValidator.getAddress.call(messageHash, signature)
+    expect(addr).to.be.equal('0xdcc703c0E500B653Ca82273B7BFAd8045D85a470')
+
+    const res = await addressValidator.compareAddressWithZeroPK.call(messageHash, signature)
+    expect(res).to.be.equal(false)
+  })
+
+  it('should return true on check with data signed with zero with message hash from tx', async function () {
```

Comment:
Whats the difference between this test and the one above?

Reply:
The first one is a test from RSKJ, the messageHash is hardcoded and it does not says what the original data is.

The second test is a hash obtained manually from a transaction with relay data